src/test/scala/chiselTests/experimental/DataView.scala

```diff
@@ -54,6 +54,26 @@ object FlatDecoupledDataView {
   implicit val view2 = view.invert(_ => new FlatDecoupled)
 }
 
+object ValidExtensions {
+  implicit def view[T <: Data] = DataView[T, Valid[T]](
+    x => Valid(x.cloneType), // Valid will strip direction with `Output(...)` anyway
+    _ -> _.bits,
+    (_, v) => true.B -> v.valid
+  )
+}
```

Comment:
Should this be a public utility?

Reply:
Probably, we need to figure out a general coding style for such methods. I think it should probably be a factory method on `object Valid` rather than making people call `.viewAs[Valid[CorrectType]]`. The problem is that the natural answer would be `Valid.apply`, but that is obviously taken by the factory for making types rather than values. It's sort of the "VecInit" pattern (but obviously that name is not the best), and if we can come up with a decent "standard", I think this would be a great example of it.